File: src/main/kotlin/no/fintlabs/linkwalker/client/ClientEvent.java
package no.fintlabs.linkwalker.client;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.springframework.util.StringUtils;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class ClientEvent {

    private Client client;
    private String orgId;
    private Operation operation;
    private String errorMessage;


    public boolean hasError() {
        return StringUtils.hasText(errorMessage);
    }

    @JsonIgnore
    public String getOrganisationObjectName() {
        return orgId.replaceAll("\\.", "_");
    }

    public enum Operation {
        CREATE,
        READ,
        UPDATE,
        DELETE
    }

}

File: src/main/kotlin/no/fintlabs/linkwalker/client/ClientEvent.kt
package no.fintlabs.linkwalker.client

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.util.StringUtils

class ClientEvent(
    val client: Client? = null,
    val orgId: String? = null,
    val operation: Operation? = null,
    val errorMessage: String? = null
) {

    fun hasError(): Boolean {
        return StringUtils.hasText(errorMessage)
    }

    @JsonIgnore
    fun getOrganisationObjectName(): String {
        return orgId!!.replace("\\.".toRegex(), "_")
    }

    enum class Operation {
        CREATE,
        READ,
        UPDATE,
        DELETE
    }

}